packages/code-infra/src/cli/cmdPublish.mjs

```diff
@@ -341,3 +399,130 @@ export default /** @type {import('yargs').CommandModule<{}, Args>} */ ({
     console.log('\n🏁 Publishing complete!');
   },
 });
+
+const WORKFLOW_PATH = 'workflows/publish.yml';
+const PUBLISH_WORKFLOW_ID = `.github/${WORKFLOW_PATH}`;
+
+/**
+ * @param {Omit<Args, 'ci'>} opts
+ */
+async function triggerLocalGithubPublishWorkflow(opts) {
+  console.log(`🔍 Checking if there are new packages to publish in the workspace...`);
+  const newPackages = await getWorkspacePackages({ nonPublishedOnly: true });
+  if (newPackages.length) {
+    console.warn(
+      `⚠️  Found new packages that should be published to npm first before triggering a release:
+  * ${newPackages.map((pkg) => pkg.name).join('  * ')}
+Please run the command "${chalk.bold('pnpm code-infra publish-new-package')}" first to publish and configure npm.`,
+    );
+    return;
+  }
+  console.log('✅ No new packages found, proceeding...');
+  const repoInfo = await getRepositoryInfo(['upstream', 'origin']);
+  console.log(`📂 Repository: ${repoInfo.owner}/${repoInfo.repo}`);
+  const params = {
+    owner: repoInfo.owner,
+    repo: repoInfo.repo,
+    workflow_id: PUBLISH_WORKFLOW_ID,
+  };
+  const octokit = getOctokit();
+
+  try {
+    const sha = opts.sha || (await determineGitSha(octokit, repoInfo));
+    if (!sha) {
+      console.error('❌🚨 No commit SHA provided, cannot proceed.');
+      return;
+    }
+
+    const res = await octokit.actions.createWorkflowDispatch({
+      ...params,
+      ref: 'master',
+      inputs: {
+        sha,
+        'dry-run': opts['dry-run'] ? 'true' : 'false',
+        'github-release': opts['github-release'] ? 'true' : 'false',
+        'dist-tag': opts.tag,
+      },
+    });
+    if (res.status > 204) {
```

Comment:
Doesn't `@octokit/rest` throw when internal requests fail?

Reply:
Removed the check.